sphinx_gallery/gen_rst.py

```diff
@@ -707,15 +721,20 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
 
     save_thumbnail(image_path_template, src_file, file_conf, gallery_conf)
 
+    peak_mem = max(memory_measurements)
     time_m, time_s = divmod(time_elapsed, 60)
     example_nb = jupyter_notebook(script_blocks)
     save_notebook(example_nb, replace_py_ipynb(example_file))
     with codecs.open(os.path.join(target_dir, base_image_name + '.rst'),
                      mode='w', encoding='utf-8') as f:
         if time_elapsed >= gallery_conf["min_reported_time"]:
             example_rst += ("**Total running time of the script:**"
-                            " ({0: .0f} minutes {1: .3f} seconds)\n\n".format(
-                                time_m, time_s))
+                            " ({0: .0f} minutes {1: .3f} seconds)\n\n"
+                            .format(time_m, time_s))
+        if PROFILE_MEMORY:
+            example_rst += ("**Peak memory usage:** {2: .3f}MB\n\n"
+                            .format(time_m, time_s, peak_mem))
```

Comment:
Why have `time_m` and `time_s` here at all rather than just using `{0:` and one `arg`?

Also is it worth adapting `sizeof_fmt` from [here](https://github.com/mne-tools/mne-python/blob/master/mne/utils.py#L2027) so that large numbers of MB get rendered as GB? You could just look for `// 1000 > 0` and format as `GB` directly, but I'm thinking such a function could also be useful elsewhere eventually.

Reply:
Ah that was a mistake, sorry.

I'll also figure out how to add that option, thanks for the comments!